Modules/System Tests/SharePoint Connector/src/SharePointTestLibrary.Codeunit.al

```diff
@@ -0,0 +1,981 @@
+        ResponseContent += '}';
+        ResponseContent += '},';
+        ResponseContent += '"ListItemAllFields":{';
+        ResponseContent += '"__deferred":{';
+        ResponseContent += '"uri":"{baseUrl}_api/Web/GetFolderByServerRelativePath(decodedurl=''{parentUrl}/Lists/Test%20Documents/Attachments/TestSubfolder'')/ListItemAllFields"';
+        ResponseContent += '}';
+        ResponseContent += '},';
+        ResponseContent += '"ParentFolder":{';
+        ResponseContent += '"__deferred":{';
+        ResponseContent += '"uri":"{baseUrl}_api/Web/GetFolderByServerRelativePath(decodedurl=''{parentUrl}/Lists/Test%20Documents/Attachments/TestSubfolder'')/ParentFolder"';
+        ResponseContent += '}';
+        ResponseContent += '},';
+        ResponseContent += '"Properties":{';
+        ResponseContent += '"__deferred":{';
+        ResponseContent += '"uri":"{baseUrl}_api/Web/GetFolderByServerRelativePath(decodedurl=''{parentUrl}/Lists/Test%20Documents/Attachments/TestSubfolder'')/Properties"';
+        ResponseContent += '}';
+        ResponseContent += '},';
+        ResponseContent += '"StorageMetrics":{';
+        ResponseContent += '"__deferred":{';
+        ResponseContent += '"uri":"{baseUrl}_api/Web/GetFolderByServerRelativePath(decodedurl=''{parentUrl}/Lists/Test%20Documents/Attachments/TestSubfolder'')/StorageMetrics"';
+        ResponseContent += '}';
+        ResponseContent += '},';
+        ResponseContent += '"Folders":{';
+        ResponseContent += '"__deferred":{';
+        ResponseContent += '"uri":"{baseUrl}_api/Web/GetFolderByServerRelativePath(decodedurl=''{parentUrl}/Lists/Test%20Documents/Attachments/TestSubfolder'')/Folders"';
+        ResponseContent += '}';
+        ResponseContent += '},';
+        ResponseContent += '"Exists":true,';
+        ResponseContent += '"IsWOPIEnabled":false,';
+        ResponseContent += '"ItemCount":0,';
+        ResponseContent += '"Name":"TestSubfolder",';
+        ResponseContent += '"ProgID":null,';
+        ResponseContent += '"ServerRelativeUrl":"{parentUrl}/Lists/Test Documents/Attachments/TestSubfolder",';
+        ResponseContent += '"TimeCreated":"2022-07-15T20:40:25Z",';
+        ResponseContent += '"TimeLastModified":"2022-07-15T20:40:25Z",';
+        ResponseContent += '"UniqueId":"922ba9b1-c26d-417f-8ecb-79d71ebfa22e",';
+        ResponseContent += '"WelcomePage":""';
+        ResponseContent += '}';
+        ResponseContent += '}';
+
+        ResponseContent := ResponseContent.Replace('{baseUrl}', BaseUrl).Replace('{parentUrl}', ParentUrl);
+        SharePointOperationResponse.SetHttpResponse(ResponseContent, HttpHeaders, 201, true, 'CREATED');
+    end;
+
+    local procedure AddFileToFolderTestResponse(var SharePointOperationResponse: Codeunit "SharePoint Operation Response"; BaseUrl: Text; ParentUrl: Text)
+    var
+        HttpHeaders: HttpHeaders;
+        ResponseContent: Text;
```

Comment:
Suggestion: please use `TextBuilder` here and in other procedures in this codeunit.

Reply:
Done